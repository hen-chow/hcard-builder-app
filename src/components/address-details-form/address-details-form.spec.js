import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AddressDetailsForm from './address-details-form';

describe('<AddressDetailsForm />', () => {
  it('invokes the provided onInputChange callback when input changes', () => {
    const onInputChangeCallback = jest.fn();
    const addressDetailsForm = shallow(<AddressDetailsForm onInputChange={onInputChangeCallback} />);
    
    addressDetailsForm.find('#suburbInput').simulate('change', { target: { value: 'Hello' } });
    
    expect(onInputChangeCallback.mock.calls.length).toEqual(1);
    expect(onInputChangeCallback.mock.calls[0][0]).toEqual({ target: { value: 'Hello' } });
  });
  
  it('renders the expected HTML', () => {
    const onInputChangeCallback = jest.fn();
    const addressDetailsForm = renderer
      .create(<AddressDetailsForm onInputChange={onInputChangeCallback} />)
      .toJSON();

    expect(addressDetailsForm).toMatchSnapshot();
  });
});
