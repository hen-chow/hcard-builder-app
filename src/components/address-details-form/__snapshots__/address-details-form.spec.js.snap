// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressDetailsForm /> renders the expected HTML 1`] = `
<div
  className="address-details-form"
>
  <span
    className="form__title"
  >
    Address
  </span>
  <hr
    className="line line--small"
  />
  <form>
    <div
      className="form__row"
    >
      <div
        className="form__field form__field--left"
      >
        <label
          htmlFor="streetNumberInput"
        >
          House Name or #
        </label>
        <input
          id="streetNumberInput"
          name="streetNumber"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
      <div
        className="form__field form__field--right"
      >
        <label
          htmlFor="streetNameInput"
        >
          Street
        </label>
        <input
          id="streetNameInput"
          name="streetName"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__field form__field--left"
      >
        <label
          htmlFor="suburbInput"
        >
          Suburb
        </label>
        <input
          id="suburbInput"
          name="suburb"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
      <div
        className="form__field form__field--right"
      >
        <label
          htmlFor="stateInput"
        >
          State
        </label>
        <input
          id="stateInput"
          name="state"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
    </div>
    <div
      className="form__row"
    >
      <div
        className="form__field form__field--left"
      >
        <label
          htmlFor="postcode"
        >
          Postcode
        </label>
        <input
          id="postcode"
          name="postcode"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
      <div
        className="form__field form__field--right"
      >
        <label
          htmlFor="countryInput"
        >
          Country
        </label>
        <input
          id="countryInput"
          name="country"
          onChange={[MockFunction]}
          type="text"
        />
      </div>
    </div>
  </form>
</div>
`;
